backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -797,20 +798,14 @@ internal class CAdapterGenerator(val context: Context) : DeclarationDescriptorVi
     private fun defineUsedTypes(scope: ExportedElementScope, indent: Int) {
         val set = mutableSetOf<ClassDescriptor>()
         defineUsedTypesImpl(scope, set)
-        // Add nullable primitives.
-        predefinedTypes.forEach {
-            val nullableIt = it.makeNullable()
+        val types = predefinedTypes + set
+                .map {  if (it.isInline) it.underlyingRepresentation()!!.type else it.defaultType }
+                .filter{isMappedToReference(it)}
+        val nullableTypes = types.map { it.makeNullable() }.toSet()
+        nullableTypes.forEach {
```

Comment:
Does this work for 
```kotlin
inline class Foo(val value: Any?)
inline class Bar(val value: Foo)

fun takeNullableBar(nullableBar: Bar?) {}
```
?

Reply:
I've added support inline and recursive inline classes for arguments and return types.

Reply:
Please add the code above to the test suite.